File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        operators = {"+" : add, "-" : sub, "*" : mul, "/" : truediv}
        s = []
        
        for t in tokens:
            if t in operators:
                b, a = s.pop(), s.pop()
                t = operators[t](a, b)
            s.append(int(t))
            
        return s[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        long a, b;
        
        for(auto t: tokens) {
            if (t == "+" or t == "-" or t == "*" or t == "/") {
                b = s.top(); s.pop();
                a = s.top(); s.pop();
                if(t == "+") 
                    s.push(a + b);
                else if(t == "-")
                    s.push(a - b);
                else if(t == "*")
                    s.push(a*b);
                else
                    s.push(a/b);
            }
            else s.push(stoi(t));
        }
        return s.top();
    }
};

